client/lxd_containers.go

```diff
@@ -71,7 +71,11 @@ func (r *ProtocolLXD) CreateContainer(container api.ContainersPost) (*Operation,
 	}
 
 	// Send the request
-	op, _, err := r.queryOperation("POST", "/containers", container, "")
+	path := "/containers"
+	if r.targetNode != "" {
+		path += fmt.Sprintf("?targetNode=%s", r.targetNode)
+	}
```

Comment:
Yeah I considered that, but for now I wanted to be really clear about what endpoints support a targetNode param, and not append it blindly. Ideally we should have some more declarative machinery that lets you tag an API as supporting that and adding the parameter if so. Considering that we already have a lot of duplicated boilerplate in the code, I'd suggest to leave this as it is, and perhaps tackle it as part of a broader normalization effort.

Reply:
Can we do the enforcement on the server side instead, so effectively add a variable to Command on the server to indicate whether target is supported for that API endpoint, then if we get a request to any endpoint that doesn't support it, we immediately return an error instead of calling the target function.

That would let us always set target in the client code whenever UseTarget() has been used and force whoever uses the Go client to only use UseTarget() for actions that do support targeting, keeping everything clean.

Reply:
It still seems sub-optimal to me to blindly append a query parameter that is not going to be used (and add ad-hoc server checks that that you're not passing it when you're not supposed to). I'd rather leave this code like this for now.